packages/scene-composer/tests/utils/dataBindingUtils.spec.ts

```diff
@@ -276,7 +276,7 @@ describe('ruleEvaluator', () => {
   });
 
   it('should property names with special characters should work correctly', async () => {
-    const key1 = 'my-temperature';
+    const key1 = 'my-temperature-prop';
     const key2 = 'myRPM';
```

Comment:
what happens when customer uses underscore and special characters in the propertyId? Will rule still work as expected?

Reply:
underscore won't be touched and will still work as expected. hyphen is the only special char in property name and is fixed now

Reply:
Confirmed from offline conversation. The change will work the following cases 
no dash in the propertyId
Multiple dash in the propertyId
single dash in the propertyId